processor/elasticinframetricsprocessor/processor.go

```diff
@@ -57,16 +59,58 @@ func newProcessor(set processor.Settings, cfg *Config) *ElasticinframetricsProce
 
 // processMetrics processes the given metrics and applies remappers if configured.
 func (p *ElasticinframetricsProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
-	for i := 0; i < md.ResourceMetrics().Len(); i++ {
-		resourceMetric := md.ResourceMetrics().At(i)
+	for resIndex := range md.ResourceMetrics().Len() {
+		resourceMetric := md.ResourceMetrics().At(resIndex)
 		rm := resourceMetric.Resource()
-
-		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
-			scopeMetric := resourceMetric.ScopeMetrics().At(j)
+		for scopeIndex := range resourceMetric.ScopeMetrics().Len() {
+			scopeMetric := resourceMetric.ScopeMetrics().At(scopeIndex)
 			for _, r := range p.remappers {
 				r.Remap(scopeMetric, scopeMetric.Metrics(), rm)
 			}
 		}
 	}
+	// drop_original=true will keep only the metrics that have been remapped based on the presense of OTelRemappedLabel label.
+	// See  https://github.com/elastic/opentelemetry-lib/blob/6d89cbad4221429570107eb4a4968cf8a2ff919f/remappers/common/const.go#L31
+	if p.cfg.DropOriginal {
```

Comment:
Will the logic be covered by some unit tests? Even if this component will be eventually be removed when the time will be to only support otel native data, however for the following releases it seems that we will be relying on this hence it would make a lot of sense to cover the implementation with some unit tests to ensure the functionality isn't/won't be broken.

Reply:
Added in https://github.com/elastic/opentelemetry-collector-components/pull/147/commits/1af050b858797a5424899d21712cdaebcee3c85c